actors/runtime/src/main/java/cloud/orbit/actors/runtime/Hosting.java

```diff
@@ -254,10 +260,11 @@ private void onClusterViewChanged(final Collection<NodeAddress> nodes)
         consistentHashNodeTree = newHashes;
         updateServerNodes();
 
+        Collection<NodeAddress> values = localAddressCache.asMap().values();
         for (NodeInfo info : oldNodes.values())
         {
             // clear local cache
-            localAddressCache.values().remove(info.address);
+            values.remove(info.address);
```

Comment:
Hmm, nice catch. I think you are correct, I doubt the collection returned by a map is smart enough to act like a set, and remove on a collection only removes a single element.

It seems like we should just build a collection based on oldNodes.values() with a list of all the addresses and then use removeAll on localAddressCache.values().